tests/test_windowed_forecast.py

```diff
@@ -107,3 +111,114 @@ def test_average_intensity_now():
             ) / window_size
         )
         assert (result == expected)
+
+
+def test_average_intensity_with_offset():
+    # Case where job start and end time are not colocated with data
+    # carbon intensity data points. In this case cats interpolate the
+    # intensity value at beginning and end of each potential job
+    # duration window.
+    CI_forecast = [
+        CarbonIntensityPointEstimate(datetime(2023,1,1,8,30), value=26),
+        CarbonIntensityPointEstimate(datetime(2023,1,1,9,0), value=40),
+        CarbonIntensityPointEstimate(datetime(2023,1,1,9,30), value=50),
+        CarbonIntensityPointEstimate(datetime(2023,1,1,10,0), value=60),
+        CarbonIntensityPointEstimate(datetime(2023,1,1,10,30), value=25),
+    ]
+    duration = 70  # in minutes
+    # First available data point is for 08:00 but the job
+    # starts 15 minutes later.
+    job_start = datetime.fromisoformat("2023-01-01T08:45")
+    result = WindowedForecast(CI_forecast, duration, start=job_start)[1]
```

Comment:
It might be me misunderstanding the role of `start` in `WindowedForecast`: how come the two rows below don't yield the same values despite being the same time windows?
```
result = WindowedForecast(CI_forecast, duration, start=datetime.fromisoformat("2023-01-01T08:45"))[1]
>>> CarbonIntensityAverageEstimate(start=datetime.datetime(2023, 1, 1, 9, 15), end=datetime.datetime(2023, 1, 1, 10, 25), value=49.970238095238095)

result2 = WindowedForecast(CI_forecast, duration, start=datetime.fromisoformat("2023-01-01T09:15"))[0]
>>> CarbonIntensityAverageEstimate(start=datetime.datetime(2023, 1, 1, 9, 15), end=datetime.datetime(2023, 1, 1, 10, 25), value=49.75595238095238)
```

Reply:
Another good catch: that's because the code in `__getitem__` assumes that the job start time (`09:15` in the above) is always in between the first and second data points. I think that's always true in the context of cats since data from the API always starts at the first available point _before_ the time of the request.

Reply:
But I realise now that it might be easy to get rid of this assumption: just trim the `data` passed to `WindowedForecast.__init__` removing all data points before the passed `start` time, except the point directly before. I'll give that a try later.

Reply:
I just made that change. If I'm correct it should work as you (reasonably) expected above.